luigi/parameter.py

```diff
@@ -343,7 +343,7 @@ def parse(self, x):
     def _warn_on_wrong_param_type(self, param_name, param_value):
         if self.__class__ != OptionalParameter:
             return
-        if not isinstance(param_value, six.string_types) and param_value is not None:
+        if not isinstance(param_value, six.string_types) and not (param_value is None and self._default != _no_value):
```

Comment:
could this type of parameter have a default value but get passed a None, which might now cause this to complain?

Reply:
I’m not sure I follow. I don’t think this change can make _new_ complaints. But are you saying that this is `OptionalParameter`, so it must have a default, so we shouldn’t even bother with the `and self._default != _no_value`, because `None` is always a valid value?

Reply:
If you had task `SomeTask` with
`some_param = luigi.OptionalParameter(default="a string")`
couldn't this complain if you instantiated it as `SomeTask(some_param=None)`?
i may be incorrect in my assumptions about where this gets applied.